http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/HTTPSClientTestCase.java

```diff
@@ -0,0 +1,96 @@
+import org.wso2.carbon.transport.http.netty.contract.HttpClientConnector;
+import org.wso2.carbon.transport.http.netty.contract.HttpResponseFuture;
+import org.wso2.carbon.transport.http.netty.contract.HttpWsConnectorFactory;
+import org.wso2.carbon.transport.http.netty.contractimpl.HttpWsConnectorFactoryImpl;
+import org.wso2.carbon.transport.http.netty.message.HTTPCarbonMessage;
+import org.wso2.carbon.transport.http.netty.message.HTTPConnectorUtil;
+import org.wso2.carbon.transport.http.netty.util.TestUtil;
+import org.wso2.carbon.transport.http.netty.util.server.HttpsServer;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+
+/**
+ * Tests for HTTPS client connector
+ */
+public class HTTPSClientTestCase {
+
+    private final Logger log = LoggerFactory.getLogger(HTTPSClientTestCase.class);
+
+    private HttpsServer httpsServer;
+    private HttpClientConnector httpClientConnector;
+//    private int port = 9092;
+    private String testValue = "Test Message";
+
+    @BeforeClass
+    public void setup() {
+        TransportsConfiguration transportsConfiguration =
+                TestUtil.getConfiguration("/simple-test-config" + File.separator + "netty-transports.yml");
+        httpsServer = TestUtil.startHttpsServer(TestUtil.TEST_HTTPS_SERVER_PORT, testValue, "text/plain");
+        HttpWsConnectorFactory connectorFactory = new HttpWsConnectorFactoryImpl();
+        httpClientConnector = connectorFactory.createHttpClientConnector(
+                HTTPConnectorUtil.getTransportProperties(transportsConfiguration),
+                HTTPConnectorUtil.getSenderConfiguration(transportsConfiguration, Constants.HTTPS_SCHEME));
+    }
+
+    @Test
+    public void testHttpsGet() {
+        try {
+            HTTPCarbonMessage msg = new HTTPCarbonMessage();
+            msg.setProperty("PORT", TestUtil.TEST_HTTPS_SERVER_PORT);
+            msg.setProperty("PROTOCOL", "https");
+            msg.setProperty("HOST", "localhost");
+            msg.setProperty("HTTP_METHOD", "GET");
+            msg.setEndOfMsgAdded(true);
+
+            CountDownLatch latch = new CountDownLatch(1);
+            HTTPSConnectorListener listener = new HTTPSConnectorListener(latch);
+            HttpResponseFuture responseFuture = httpClientConnector.send(msg);
+            responseFuture.setHttpConnectorListener(listener);
+
+            latch.await(5, TimeUnit.SECONDS);
+
+            HTTPCarbonMessage response = listener.getHttpResponseMessage();
+            assertNotNull(response);
+            String result = new BufferedReader(new InputStreamReader(response.getInputStream()))
+                    .lines().collect(Collectors.joining("\n"));
+
+            assertEquals(testValue, result);
```

Comment:
Looks like the variables are switched.
public static void assertEquals(Object actual, Object expected) {